presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -418,24 +509,56 @@ private void flushStripe(FlushReason flushReason)
         columnEncodings.put(0, new ColumnEncoding(DIRECT, 0));
         columnStatistics.put(0, new ColumnStatistics((long) stripeRowCount, 0, null, null, null, null, null, null, null, null));
 
-        // add footer
-        StripeFooter stripeFooter = new StripeFooter(allStreams, toDenseList(columnEncodings, orcTypes.size()));
+        Map<Integer, ColumnEncoding> unencryptedColumnEncodings = columnEncodings.entrySet().stream()
```

Comment:
@zzhao0  this is where we put the encodings and streams into the stripe encryption groups

Reply:
This logic looks right to me at first glance. Will take a closer look later.